yb_build_thirdparty_main.py

```diff
@@ -958,6 +958,99 @@ def get_openssl_related_cmake_args(self):
+        self.tp_installed_dir = os.path.join(self.tp_dir, 'installed')
+
+    def check_lib_deps(self, file_path, cmdout):
+        status = True
+        for line in cmdout.splitlines():
+            # Dependent libs indented with tab
+            if line.startswith('\t'):
+                if not re.match(self.okay_paths,line):
+                    if status:
+                        log(file_path + ":")
+                    log("Bad path: %s", line)
+                    status = False
+        return status
+
+    # overridden in platform specific classes
+    def good_libs(self, file_path):
+        pass
+
+    def run(self):
+        log("Scanning installed executables and libraries...")
+        test_pass = True
+        # files to examine are much reduced if we look only at bin and lib directories
+        dir_pattern = '^(lib|libcxx|[s]bin)$'
+        dirs = [os.path.join(self.tp_installed_dir, type) for type in BUILD_TYPES]
+        for installed_dir in dirs:
+            with os.scandir(installed_dir) as candidate_dirs:
+                for candidate in candidate_dirs:
+                    if re.match(dir_pattern, candidate.name):
+                        examine_path = os.path.join(installed_dir, candidate.name)
+                        for dirpath, dirnames, files in os.walk(examine_path):
+                            for file_name in files:
+                                full_path = os.path.join(dirpath, file_name)
+                                if os.path.islink(full_path):
+                                    continue
+                                if not self.good_libs(full_path):
+                                    test_pass = False
+        if not test_pass:
+            fatal(f"Found problematic library dependencies, using tool: {self.tool}")
+        else:
+            log("No problems found with library dependencies.")
+
+
+class LibTestMac(LibTestBase):
+    def __init__(self):
+        super().__init__()
+        self.tool = "otool -L"
+        path_regex = ["^\t/usr/",
+                       "^\t/System/Library/",
+                       "^\t@rpath",
+                       "^\t@loader_path",
+                       f"^\t{self.tp_dir}"]
+        self.okay_paths = "|".join(path_regex)
+
+    def good_libs(self, file_path):
+        libout = subprocess.check_output(['otool', '-L', file_path]).decode('utf-8')
+        if 'is not an object file' in libout:
+            return True
+        return self.check_lib_deps(file_path, libout)
+
+
+class LibTestLinux(LibTestBase):
+    def __init__(self):
+        super().__init__()
+        self.tool = "ldd"
+        path_regex = ["^\tlinux-vdso",
+                       "^\t/lib64/",
+                       "^\t/opt/yb-build/brew/linuxbrew",
+                       "^\tstatically linked",
+                       "^.* => /lib64/",
+                       "^.* => /opt/yb-build/brew/linuxbrew",
+                       f"^.* => {self.tp_dir}"]
```

Comment:
It would probably be better to assign this list itself to a field like `self.okay_path_patterns` in the derived class constructor, and then join it with `|` and compile the regex once in the base class's `run()` method.